File: Linked List/intersectionOfLinkedList.py
class Solution:
    # @param A : head node of linked list
    # @param B : head node of linked list
    # @return the head node in the linked list
    def length(self, A):
        n = 0
        while (A):
            n += 1
            A = A.next
        return n
    def getIntersectionNode(self, A, B):
        m = self.length(A)
        n = self.length(B)
        if (n > m):
            return self.getIntersectionNode(B,A)
        diff = abs(m - n)
        for _ in range(diff):
            A = A.next
        while (A and B):
            if (A is B):
                return A
            A = A.next
            B = B.next
        return None

File: Linked List/intersectionOfLinkedList.js
function ListNode(val) {
  this.val = val;
  this.next = null;
}

var length = function(A) {
  var count = 0;
  while (A) {
    count++;
    A = A.next;
  }
  return count;
};
var a = new ListNode(1, c);
var b = new ListNode(3, c);
var c = new ListNode(2);
var getIntersectionNode = function(headA, headB) {
  var m = length(headA);
  var n = length(headB);
  if (n > m) {
    return getIntersectionNode(headB, headA);
  }

  var diff = Math.abs(m - n);
  for (var i = 0; i < diff; i++) {
    headA = headA.next;
  }
  while (headA && headB) {
    if (headA === headB) {
      return headA;
    }
    headA = headA.next;
    headB = headB.next;
  }

  return null;
};
console.log(getIntersectionNode(a, c));
